corehq/apps/hqwebapp/static/hqwebapp/js/daterangepicker.config.js

```diff
@@ -37,8 +37,11 @@ $.fn.createDateRangePicker = function (
     };
     var hasStartAndEndDate = !_.isEmpty(startdate) && !_.isEmpty(enddate);
     if (hasStartAndEndDate) {
-        config.startDate = new Date(startdate);
-        config.endDate = new Date(enddate);
+        // Parse dates as local dates to avoid timezone conversion
+        var startParts = startdate.split('-');
+        var endParts = enddate.split('-');
+        config.startDate = new Date(startParts[0], startParts[1] - 1, startParts[2]);
+        config.endDate = new Date(endParts[0], endParts[1] - 1, endParts[2]);
```

Comment:
Are `startParts[1] - 1` and `endParts[1] - 1` both subtracting a number (1) from a string? (checks some string - number arithmetic in console) I guess that works in JavaScript...

Is it true that `startdate` and `enddate` will always be strings formatted as YYYY-MM-DD?

Reply:
Yes to both questions.
[The template](https://github.com/dimagi/commcare-hq/blob/885a4dbfc4d587619158d8fd4971229588d541c1/corehq/apps/reports/templates/reports/filters/bootstrap3/datespan.html#L14-L15) is using Django's |date:"Y-m-d" filter, which always produces strings in the YYYY-MM-DD format.
Also pushed a commit to explicitly convert string to int: 6bbd6a39c68b40a76eb83bc767f56f805aa14de8